.github/workflows/gh-pages.yml

```diff
@@ -4,21 +4,37 @@ on:
   # Runs every time master branch is updated
   push:
     branches: ["master"]
+  # Runs every time a PR is open against master
+  pull_request:
+    branches: ["master"]
   workflow_dispatch:
 
+concurrency:
+  # Prevent 2+ copies of this workflow from running concurrently
+  group: e3sm-docs-action
```

Comment:
To get really fancy, you change this to
```
group: ${{ github.ref }}
```
so that different PRs can still build at the same time (they deploy to different preview pages anyways)

Reply:
You can also add
```
   cancel-in-progress: true
```
to the concurrency section, since we prob don't care about finishing to build docs if a commit is pushed...